src/util/kdtree.jl

```diff
@@ -38,8 +40,15 @@ Construct a K-D tree (`KDTree`) representation of a list of
 When searching the tree, shapes that appear earlier in `s`
 take precedence over shapes that appear later.
 """
+
 function KDTree(s::AbsVec{S}) where {K,S<:Shape{K}}
-    (length(s) ≤ 4 || K == 0) && return KDTree{K,S}(s)
+    # If no list of indicies is provided, simply enumerate by the number of
+    # shapes in `s`.
+    return KDTree(s,collect(eachindex(s)))
+end
+
+function KDTree(s::AbsVec{S}, s_index::Vector{<:Int}) where {K,S<:Shape{K}}
```

Comment:
`Vector{<:Int}` is equivalent to `Vector{Int}` since `Int` is a concrete type (which cannot have subtypes in Julia — all concrete types are final).  If you want to support arbitrary vectors of integers here it shoulod be:
```suggestion
function KDTree(s::AbsVec{S}, s_index::AbstractVector{<:Integer}) where {K,S<:Shape{K}}
```
(On a separate note, I think we should get rid of the aliases like `AbsVec` that @wsshin defined — these are not idiomatic and make the code harder to read for other Julia programmers.  But that can be a separate PR.)

Reply:
> On a separate note, I think we should get rid of the aliases like AbsVec

yep, I agree.